http-server/src/server.rs

```diff
@@ -469,3 +375,110 @@ fn is_json(content_type: Option<&hyper::header::HeaderValue>) -> bool {
 		_ => false,
 	}
 }
+
+/// Process a verified request, it implies a POST request with content type JSON.
+async fn process_validated_request(
```

Comment:
move to a separate function to decrease the indentation level and become more readable...